django/db/models/fields/__init__.py

```diff
@@ -896,6 +896,9 @@ def value_from_object(self, obj):
         """Return the value of this field in the given model instance."""
         return getattr(obj, self.attname)
 
+    def slice_expression(self, expression, start, length):
```

Comment:
As @adamchainz suggested, I let this method return  `None` if slicing is not applicable to the field. An `is_sliceable` flag in addition would probably be more clear, however it would also a bit redundant.

Reply:
The second argument isn't `length` but the `end`, so rename it here and in every other case?

Reply:
My bad it is the `length` after `SliceableF` calculates it.